sdk/core/azure-core/src/datetime.cpp

```diff
@@ -4,869 +4,832 @@
-  int secondsInt = static_cast<int>(secondsLeft - static_cast<int64_t>(year4) * SecondsIn4Years);
+constexpr int16_t DayOfYear(int16_t year, int8_t month, int8_t day)
+{
+  int16_t daysInPrevMonths = 0;
+  for (int i = 1; i < month; ++i)
+  {
+    daysInPrevMonths += MaxDaysPerMonth[i - 1];
+  }
 
-  int year1 = secondsInt / SecondsInYear;
-  if (year1 == 4)
+  if (month > 2 && !IsLeapYear(year))
   {
-    // this is the last day in a leap year
-    year1 = 3;
+    daysInPrevMonths -= 1;
   }
-  secondsInt -= year1 * SecondsInYear;
 
-  return {year400 * 400 + year100 * 100 + year4 * 4 + year1, secondsInt};
+  return daysInPrevMonths + day;
 }
 
-constexpr bool IsLeapYear(int year)
+constexpr int32_t DaySinceEpoch(int16_t year, int8_t month, int8_t day)
 {
-  return (year % 4 == 0 && (year % 100 != 0 || year % 400 == 0));
+  int16_t const priorYear = year - 1;
+  auto const leapYears = LeapYarsSinceEpoch(priorYear);
+  auto const nonLeapYears = priorYear - leapYears;
+  return (nonLeapYears * 365) + (leapYears * 366) + DayOfYear(year, month, day);
 }
 
-// The following table assumes no leap year; leap year is added separately
-constexpr unsigned short const CumulativeDaysToMonth[12] = {
-    0, // Jan
-    31, // Feb
-    59, // Mar
-    90, // Apr
-    120, // May
-    151, // Jun
-    181, // Jul
-    212, // Aug
-    243, // Sep
-    273, // Oct
-    304, // Nov
-    334 // Dec
-};
-
-constexpr unsigned short const CumulativeDaysToMonthLeap[12] = {
-    0, // Jan
-    31, // Feb
-    60, // Mar
-    91, // Apr
-    121, // May
-    152, // Jun
-    182, // Jul
-    213, // Aug
-    244, // Sep
-    274, // Oct
-    305, // Nov
-    335 // Dec
-};
-
-constexpr char const dayNames[] = "Sun\0Mon\0Tue\0Wed\0Thu\0Fri\0Sat";
-constexpr char const monthNames[] = "Jan\0Feb\0Mar\0Apr\0May\0Jun\0Jul\0Aug\0Sep\0Oct\0Nov\0Dec";
-
-} // namespace
+constexpr int8_t GetDayOfWeek(int16_t year, int8_t month, int8_t day)
+{
+  return DaySinceEpoch(year, month, day) % 7;
+}
 
-std::string DateTime::GetString(DateFormat format, TimeFractionFormat fractionFormat) const
+constexpr int8_t WeekDayMonthDayOfYear(int8_t* month, int8_t* day, int16_t year, int16_t dayOfYear)
 {
-  static constexpr auto const EndOfYear9999 = 2650467743999999999LL;
-  if (m_since1601.count() > EndOfYear9999)
+  auto remainder = dayOfYear;
+  for (int8_t i = 1; i <= 12; ++i)
   {
-    throw std::invalid_argument("The requested year exceeds the year 9999.");
-  }
-
-  int64_t const epochAdjusted = m_since1601.count();
-  int64_t const secondsSince1601 = epochAdjusted / Duration::period::den; // convert to seconds
-  int const fracSec = static_cast<int>(epochAdjusted % Duration::period::den);
-
-  auto const yearData = ComputeYear(secondsSince1601);
-  int const year = yearData.Year + MinYear;
-  int const yearDay = yearData.SecondsLeftThisYear / SecondsInDay;
+    int8_t const MonthDays = MaxDaysPerMonth[i - 1] - ((i == 2 && !IsLeapYear(year)) ? 1 : 0);
```

Comment:
This logic is difficult to parse and understand. Add a comment explaining what you are trying to do here.